### PR TITLE
[Core] Adding Newton-Cotes integration points for line

### PR DESCRIPTION
**Pull request by loumalouomega**
Part of the fix for #6646.

### CODE REVIEW

**Comment by loumalouomega** [comment]
@pooyan-dadvand  I'm having problem sdefining the integrations points as a static member variable of the line (non const), with the corresponding method in order to modify the extended methods so this integration points can be considered. For now I only add the integration points definition

**Comment by loumalouomega** [comment]
@Vahid-Galavi in the added test you can see how to use it

File: kratos/integration/line_newton_cotes_integration_points.h [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+
+    static SizeType IntegrationPointsNumber()
+    {
+        return 3;
+    }
+
+    static IntegrationPointsArrayType& IntegrationPoints()
+    {
+        // This is added to solve the problem of static initialization. Pooyan.
+        msIntegrationPoints[0]=IntegrationPointType(-0.5,1.333333333333333);
```
Comment (GuillermoCasas) [inline kratos/integration/line_newton_cotes_integration_points.h:48]: why not use actual fractions?

**Comment by loumalouomega** [comment]
Because it was generated with a matlab script from the Felippa webpage...

**Comment by GuillermoCasas** [comment]
On behalf of the @KratosMultiphysics/implementation-committee , this PR is ready to merge. The PR adds a new integration rule together with the appropriate test, which also serves as a test for the changes in #6745. We leave it to the @KratosMultiphysics/technical-committee to give their final decision on the topic.

**Comment by loumalouomega** [comment]
This is ready after merging #6926 @pooyan-dadvand

File: kratos/geometries/geometry.h [hunk 2]
```diff
@@ -1279,6 +1279,20 @@ class Geometry
         return result;
     }
 
+    /**
+     * @brief It returns a vector that is normal to its corresponding geometry in a given integration method
+     * @param IntegrationPointIndex index of integration point which jacobians has to be calculated in it.
+     * @param rIntegrationPointsArray The integration method which jacobians has to be calculated in its integration points.
+     * @return The normal in the given integration method
+     */
+    virtual array_1d<double, 3> Normal(
```
Comment (pooyan-dadvand) [inline kratos/geometries/geometry.h]: Why we need to have this new interface. We can clearly use the below one passing the integration point as local coordinates as you did in line 1293. So what would be the benefit?

File: kratos/geometries/geometry.h [hunk 3]
```diff
@@ -1375,6 +1389,20 @@ class Geometry
         return normal;
     }
 
+    /**
+     * @brief It computes the unit normal of the geometry in the given integration method
+     * @param IntegrationPointIndex index of integration point which jacobians has to be calculated in it.
+     * @param rIntegrationPointsArray The integration method which jacobians has to be calculated in its integration points.
+     * @return The unit normal in the given integration method
+     */
+    virtual array_1d<double, 3> UnitNormal(
```
Comment (pooyan-dadvand) [inline kratos/geometries/geometry.h]: Same as the normal

File: kratos/geometries/geometry.h [hunk 4]
```diff
@@ -2097,6 +2125,22 @@ class Geometry
...
     }
 
+    /**
+     * @brief This method provides the global coordinates corresponding to the given integration method.
+     * @param rResult The array containing the global coordinates corresponding to the given integration method.
+     * @param IntegrationPointIndex index of integration point which jacobians has to be calculated in it.
+     * @param rIntegrationPointsArray The integration method which jacobians has to be calculated in its integration points.
+     * @return An array containing the global coordinates corresponding to the given integration method.
+     */
+    virtual CoordinatesArrayType& GlobalCoordinates(
```
Comment (pooyan-dadvand) [inline kratos/geometries/geometry.h]: Again.

**Review by pooyan-dadvand — COMMENTED** [review]
To be honest, I don't see a performance benefit in creating many of these methods and I feel that we only crowding the interface by having auxiliary functions helping in one line implementation outside. 

Also, it is not coherent with the old way of working with integration points by passing them via enum. Why we cannot (or we don't want) to use the same mechanism here having enums for new integral rules?

**Comment by loumalouomega** [comment]
I clean up what you mentioned @pooyan-dadvand. On the other hand, the integration method was added after the request of @Vahid-Galavi

**Comment by pooyan-dadvand** [comment]
I completely agree with adding the integration method. 

Here, the discussion is about the interfaces and the details of the implementation. 

Same comment is also applied to the rest of the "Specialized" methods for external integration points. So let  check the other opinions. 

Also, @KratosMultiphysics/technical-committee?

**Comment by loumalouomega** [comment]
@philbucher CI is not working due to mmg...

**Comment by RiccardoRossi** [comment]
same as #6745

**Comment by loumalouomega** [comment]
I think this can be retaken @KratosMultiphysics/technical-committee

**Comment by loumalouomega** [comment]
What is the status of this¿? https://github.com/orgs/KratosMultiphysics/teams/technical-committee

**Comment by loumalouomega** [comment]
> What is the status of this¿? [github.com/orgs/KratosMultiphysics/teams/technical-committee](https://github.com/orgs/KratosMultiphysics/teams/technical-committee)

👋

**Comment by loumalouomega** [comment]
This is 4 yeras old, can we take a look?

**Comment by rubenzorrilla** [comment]
> This is 4 yeras old, can we take a look?

At first glance, I'd suggest to merge the master again and add any testing.

**Comment by loumalouomega** [comment]
> > This is 4 yeras old, can we take a look?
> 
> At first glance, I'd suggest to merge the master again and add any testing.

Done

**Comment by loumalouomega** [comment]
@rubenzorrilla This is ready

**Comment by rubenzorrilla** [comment]
For me it would be fine. As far as I understand @pooyan-dadvand's observations are no longer a problem right?

**Comment by loumalouomega** [comment]
> For me it would be fine. As far as I understand @pooyan-dadvand's observations are no longer a problem right?

Exactly

**Review by rubenzorrilla — APPROVED** [review]

**Pull request merged**